### PR TITLE
dockerng: avoid network duplication and add driver_opts

### PR DESCRIPTION
**Pull request by ghost**
### What does this PR do?

- Avoid docker network duplication by default (by setting ``check_duplicate=True``)
- Add ``driver_opts`` (or ``options`` as per API) to a dockerng network_present (state) / create_network (module).

``driver_opts`` can be used this way:

**SLS**
```
the networkabc network:
  dockerng.network_present:
    - name: networkabc
    - driver_opts:
      - com.docker.network.driver.mtu: 1450
```

**salt-call**
```
salt 'your-target-1' dockerng.create_network networkabc driver=bridge driver_opts="{'com.docker.network.driver.mtu':'1450'}"
```

### What issues does this PR fix or reference?

None was created yet, but it addresses these points:
- The docker daemon [API](https://docs.docker.com/engine/api/v1.26/#operation/NetworkCreate) [lets](https://github.com/docker/docker/issues/18864) you create two networks with the same name by default. dockermod (dockerng) should not do this by default, IMHO.
- Inability to set docker-network driver specific options. ``--opt``

### Tests written?

Yes.

### CODE REVIEW

File: salt/states/docker_network.py [hunk 1]
```diff
@@ -110,7 +129,11 @@ def present(name, driver=None, containers=None):
     else:
         try:
             ret['changes']['created'] = __salt__['docker.create_network'](
-                name, driver=driver)
+                name,
+                driver=driver,
+                driver_opts=driver_opts,
+                check_duplicate=True)
```
Comment (ticosax) [inline salt/states/docker_network.py]: check_duplicate=check_duplicate

**Review by ticosax — APPROVED** [review]
LGTM

File: salt/states/docker_network.py (hunk 1 above)
Comment (arno01) [inline salt/states/docker_network.py]: IIUC, this way ``check_duplicate`` will always be ``True`` whenever ``docker_network.present`` is called.
Seems logical to me as we don't want this state to produce a duplicated network.

Thinking further, be there some failure (race?) in salt to [whether the network already exist](https://github.com/saltstack/salt/blob/develop/salt/states/docker_network.py#L96),  this duplication may occur. So LGTM having it set to ``True`` :-)

Thank you.

Comment (cachedout) [inline salt/states/docker_network.py]: @ticosax Any response to this?

Comment (ticosax) [inline salt/states/docker_network.py]: @arno01 I understand, 
Why exposing it as an argument of the state then ?

Comment (arno01) [inline salt/states/docker_network.py]: @ticosax got it, I've updated the PR. Please let me know if everything is ok now. Thanks.

Comment (cachedout) [inline salt/states/docker_network.py]: If @ticosax OKs this change then we'll get it in. Thanks!

**Review by ticosax — APPROVED** [review]

File: salt/states/docker_network.py (hunk 1 above)
Comment (ticosax) [inline salt/states/docker_network.py]: all good. thx.

**Pull request merged**